modules/core/src/main/java/com/treasure_data/logger/sender/ExtendedPacker.java

```diff
@@ -32,36 +32,40 @@
 class ExtendedPacker {
     private static Logger LOG = Logger.getLogger(ExtendedPacker.class.getName());
 
-    private static int keySoftLimit = 256;
+    private static final int KEY_SOFT_LIMIT = 256;
+    private static final int KEY_HARD_LIMIT = 512;
 
-    private static int keyHardLimit = 512;
+    private final MessagePack msgpack;
 
     private Packer packer;
-
     private ByteArrayOutputStream out;
-
     private GZIPOutputStream gzout;
-
-    private List<String> keys = new ArrayList<String>(keySoftLimit);
-
-    private long rowSize = 0;
+    private List<String> keys;
+    private long rowSize;
 
     ExtendedPacker(MessagePack msgpack) throws IOException {
+        this.msgpack = msgpack;
+        refresh();
+    }
+
+    private void refresh() throws IOException {
         out = new ByteArrayOutputStream();
         gzout = new GZIPOutputStream(out);
```

Comment:
To get the size of compressed data, `GZipOutputStream` object wraps `ByteArrayOutputStream` object. We can get the size from `ByteArrayOutputStream#size()`.